force-app/main/default/classes/DMLOperations.cls

```diff
@@ -162,59 +196,103 @@ public with sharing class DMLOperations {
 	public static Account upsertAccount(String accountName) {
 		// Query for the Account
 		List<Account> accounts = [SELECT Id FROM Account WHERE Name = :accountName];
-
+        Account upsertAccount;
 		// if the Account exists, update the description
-
+        if(!accounts.isEmpty()) {  
+           Account existingAccount = accounts[0]; // Finding the existing account in the list by index
+           existingAccount.description = 'Updated Account'; 
+           Update existingAccount;  // Update the existing account into production
+           upsertAccount = existingAccount;
+           
+        }else{
+            Account newAccount = New Account();
+            newAccount.name = accountName;
+            newAccount.description = 'New Account';
+            Insert newAccount; // Insert the new account into production
+            upsertAccount = newAccount;
+        }
```

Comment:
but, you can do just one DML operation if you assign the account within the if/else block, and then call upsert instead of separate insert/update DML like above.